engine/channels.go

```diff
@@ -21,20 +21,22 @@ var channelRoleMap map[network.Channel]flow.RoleList
 
 // RolesByChannel returns list of flow roles involved in the channel.
 func RolesByChannel(channel network.Channel) (flow.RoleList, bool) {
-	if clusterChannel, isCluster := ClusterChannel(channel); isCluster {
-		// replaces channel with the stripped-off prefix
-		channel = clusterChannel
+	if _, isCluster := ClusterChannel(channel); isCluster {
+		return ClusterChannelRoles(), true
```

Comment:
right, but channelRoleMap doesn't make sense for cluster-based channels imo because they are unique to the chain ID. ie, the value stored in `channelRoleMap` is not actually a "real" channel

Reply:
I wonder if we can remove `ChannelSyncCluster()` and `ChannelConsensusCluster` entirely, and simply redefine `TopicFromChannel` to take in a chain ID instead? Then we'd be able to rename `syncClusterPrefix` to simply `SyncCluster` (similary for consensus) and add it to the channel role map